pkg/operator/controller/ingress/controller.go

```diff
@@ -713,6 +726,121 @@ func (r *reconciler) ensureIngressDeleted(ingress *operatorv1.IngressController)
 	return utilerrors.NewAggregate(errs)
 }
 
+func (r *reconciler) clearRouteStatusForDeletedIngress(icName string) error {
+	// Clear route status for routes that have been admitted by this ingress controller.
+	// List all routes.
+	start := time.Now()
+	routeList := &routev1.RouteList{}
+	routesCleared := 0
+	if err := r.cache.List(context.TODO(), routeList); err != nil {
+		return fmt.Errorf("failed to list all routes in order to clear route status")
+	}
+	for _, route := range routeList.Items {
+		cleared, err := r.clearRouteStatus(route, icName)
+		if cleared {
+			routesCleared++
+		}
+		if err != nil {
+			return err
+		}
+	}
+	elapsed := time.Since(start)
+	log.Info("cleared route status for ingress deletion", "Ingress Controller", icName, "Routes Status Cleared", routesCleared, "Time Elapsed", elapsed)
+	return nil
+}
+
+// Clears this route's status that is admitted by provided ingressController
+func (r *reconciler) clearRouteStatus(route routev1.Route, icName string) (bool, error) {
+	// Go through each route and clear status if admitted by this ingress controller.
+	var updated routev1.Route
+	clearedStatus := false
+	for i := range route.Status.Ingress {
+		if condition := findCondition(&route.Status.Ingress[i], routev1.RouteAdmitted); condition != nil {
+			if routerName := &route.Status.Ingress[i].RouterName; routerName != nil && *routerName == icName {
+				// Remove this status since it matches our routerName.
+				route.DeepCopyInto(&updated)
+				updated.Status.Ingress = append(route.Status.Ingress[:i], route.Status.Ingress[i+1:]...)
+				if err := r.client.Status().Update(context.TODO(), &updated); err != nil {
+					return clearedStatus, fmt.Errorf("failed to clear route status for %s with routerName %s: %v", route.Name, icName, err)
```

Comment:
Include the route's namespace to reduce ambiguity.  (It maybe `err` already includes the route's name and namespace?)  Also, it's good practice to use `%w` when wrapping error values.  
```suggestion
					return clearedStatus, fmt.Errorf("failed to clear route status of route %s/%s for router %s: %w", route.Namespace, route.Name, icName, err)
```

Reply:
Great, got it